### ISSUE TITLE
crapi/gateway-service:latest always turns unhealthy

### ISSUE THREAD

**Issue by Mesdop** [OWASP/crAPI#210]
When attempting to launch crAPI every container for it works except for crapi/gateway. It takes awhile to start then turns unhealthy instantly.

I'm using the latest stable version following the instructions in the quick-start guide. Following the commands, every service is running healthy and I can connect to them via localhost except for crapi/gateway-service. I also attempted to build it myself by cloning the repository but encounter the same issue.

![Screenshot from 2023-06-27 18-18-41](https://github.com/OWASP/crAPI/assets/135493885/61110338-07f0-4487-95e4-6495d7b40ae2)

I'm using Oracle VM Virtualbox to run Ubuntu 22.04.2 LTS 64-bit GNOME version 42.5

Running "docker-compose version" gives me:

docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.10.6
OpenSSL version: OpenSSL 3.0.2 15 Mar 2022

**Comment by wannab3hack3r** [comment]
> I solved it by removing the health checks. Remove health checks on every service along with `depends_on` part which specifies `condition: service_healthy`

how can i do that

**Comment by Mesdop** [comment]
I tried removing the health checks along with everything under `depends_on` but still haven't had success.

I'm running the docker on an Ubuntu VirtualBox and attempting to connect to it from a second VB running Kali Linux. I've gotten other dockers to run and had success connecting to them so I'm unsure why with crAPI I'm struggling.

**Comment by thebigbone** [comment]
> I tried removing the health checks along with everything under `depends_on` but still haven't had success.
> 
> I'm running the docker on an Ubuntu VirtualBox and attempting to connect to it from a second VB running Kali Linux. I've gotten other dockers to run and had success connecting to them so I'm unsure why with crAPI I'm struggling.

Did you do it for every single service? Well, I am running Ubuntu on bare metal, haven't tried it in a VM

**Comment by Mesdop** [comment]
I finally found a solution, not sure if I'm allowed to link websites/youtube channels here but this is what I did to follow the process:

This YouTube Channel perfectly explained the issue I was having with it: [https://www.youtube.com/watch?v=ByW_yobA2aM&t=5s](url)

He also has a web page where he shows all the commands needed to make it work: [https://samsclass.info/129S/proj/AP110.htm](url)

I created a new Virtual Machine running Ubuntu 18.04.6 LTS (Bionic Beaver) using the Desktop Image.
Following the steps on the web page provided, you will encounter the same issue I was running into before where crapi/gateway turns unhealthy.

Following @thebigbone instructions:
> I solved it by removing the health checks. Remove health checks on every service along with `depends_on part` which specifies `condition: service_healthy`

Makes it work.

Thanks for the help! Would of never figured out the part to remove health checks and everything online says that they're needed.

No idea why Ubuntu 18.04 works, I also tried using the docker in Kali Linux but encounter the same issue. I didn't try Windows however.
